### PR TITLE
dont print apikey in remote client toString, add hostoverride to python client

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**

### CODE REVIEW

File: node/src/remote/client.ts [hunk 1]
```diff
@@ -102,4 +102,11 @@ export class HttpLancedbClient {
     }
     return response
   }
+
+  toString (): string {
```
Comment (eddyxu) [inline node/src/remote/client.ts:106]: is this what `console.log(..)` uses?

File: python/lancedb/__init__.py [hunk 2]
```diff
@@ -55,5 +60,9 @@ def connect(
     if isinstance(uri, str) and uri.startswith("db://"):
         if api_key is None:
             raise ValueError(f"api_key is required to connected LanceDB cloud: {uri}")
-        return RemoteDBConnection(uri, api_key, region)
+        return RemoteDBConnection(uri, api_key, region, host_override)
+    if host_override is not None:
+        warnings.warn(
+            "host_override has no effect when connecting to FS based DB", Warning
```
Comment (eddyxu) [inline python/lancedb/__init__.py:66]: what is `fs based db`??  We don't need this warning, for example, api-key is not checked here as well.

**Review by eddyxu — APPROVED** [review]
Minor qq.

File: node/src/remote/client.ts (hunk 1 above)
Comment (chebbyChefNEQ) [reply node/src/remote/client.ts:106]: I think so yeah

**Pull request merged**